### ISSUE TITLE
[Feature] jsonify pytorch tensor

### ISSUE THREAD

**Issue by ProtossDragoon** [open-mmlab/mmengine#736]
### What is the feature?

Hi,

I want to jsonify a MMOCR result, formatted like this: 

```python
{'rec_texts': [ ... ], 'rec_scores': [ ... ], 'det_polygons': [ ... ], 
'det_scores': tensor(
       [0.8443, 0.8401, 0.8440, 0.8488, 0.8146, 0.6879, 0.8354, 0.8199, 0.8820,
        0.8532, 0.5870, 0.8671, 0.9009, 0.8786, 0.9155, 0.9403, 0.6059, 0.8203,
        0.8601, 0.4475])}
```

But the MMOCR API doesn't permit to pass any additional keyword argument to control MMEngine features. So I cannot pass custom json handler that handles pytorch tensor.

- https://github.com/open-mmlab/mmocr/blob/29107ef81da05b3ad0cc8847ae3b8c13501a1716/mmocr/ocr.py#L190-L222
- https://github.com/open-mmlab/mmocr/blob/29107ef81da05b3ad0cc8847ae3b8c13501a1716/mmocr/apis/inferencers/base_mmocr_inferencer.py#L123-L151

When I want to convert pytorch tensor to json,

https://github.com/open-mmlab/mmengine/blob/e98d2624151461a83fb30b420a5ecc3ef26ff17f/mmengine/fileio/handlers/json_handler.py#L9-L22 

above code should be changed like this.

```python
def set_default(obj):
    if isinstance(obj, (set, range)):
        return list(obj)
    elif isinstance(obj, np.ndarray):
        return obj.tolist()
    elif isinstance(obj, np.generic):
        return obj.item()
    elif isinstance(obj, torch.Tensor):
        return obj.tolist()
    raise TypeError(f'{type(obj)} is unsupported for json dump')
```

But I have no idea about correct modification to support this feature. Because MMEngine abstraction is the lowest level of any  other MMxxx frameworks. Could this feature be supported? or could you give me some advice?

**Comment by HAOCHENYE** [comment]
Thanks for your feedback and suggestions 🚀 ! I think it could be reasonable to handle tensor in `set_default`. We may ask for some suggestions from @zhouzaida

**Comment by ZwwWayne** [comment]
Also invite @gaotongxiao to have a look

**Comment by gaotongxiao** [comment]
This is likely an implementation flaw in XXXInferencer, whose returned elements should always be simple. I will post a quick fix to MMOCR soon.

BTW, your solution also works, but `mmengine.fileio` should not have the dependency on `torch` for the sake of import efficiency.

**Comment by zhouzaida** [comment]
MMOCR will fix this error in [#1559](https://github.com/open-mmlab/mmocr/pull/1559).
